### ISSUE TITLE
Unbound method finalize_options

### ISSUE THREAD

**Issue by vaab** [embray/d2to1#38]
Last update of travis python 2.7 virtual machines made my travis build break.

Full logs: https://travis-ci.org/0k/kids.sh/jobs/49698894

I'm using: `python setup.py install` to make it break.

Please notice that it was working yesterday (5 Feb 2015). 

Here's an excerpt of the failing Traceback::

```
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/setuptools/dist.py", line 269, in __init__
    _Distribution.__init__(self,attrs)
  File "/opt/python/2.7.9/lib/python2.7/distutils/dist.py", line 287, in __init__
    self.finalize_options()
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/setuptools/dist.py", line 325, in finalize_options
    ep.load()(self, ep.name, value)
  File "/home/travis/build/0k/kids.sh/.eggs/d2to1-0.2.11-py2.7.egg/d2to1/core.py", line 70, in d2to1
    _Distribution.finalize_options(dist)
TypeError: unbound method finalize_options() must be called with Distribution instance as first argument (got Distribution instance instead)
```

I noticed that these command where behaving also differently on python 3.4 and broke some `python setup.py` black magic. But I managed to get through, and `d2to1` was not an issue. I've seen that pip moved from version `1.5.6` to `6.0.7` since yesterday on both python 3.4 and 2.7 VMs. 

The new setuptools is `12.0.5`. Can't get the previous version. It's probably the core of the issue.

Any insights to what I could do to make these pass like before ?

**Comment by vaab** [comment]
I could reproduce the issue in a local virtualenv and installing setuptools==12.0.5 and python 2.7

Just running ``python setup.py install`` fails but on installing dependencies only.

It says: ``Running kids.test-0.0.1/setup.py -q bdist_egg --dist-dir /tmp/easy_install-K03mD7/kids.test-0.0.1/egg-dist-tmp-F6WV3l``

The dependency is using ``d2to1``.

It happens that one of my dependency has no dependency and is using ``d2to1``. Downloading the source code, then issuing ``python setup.py install`` worked without any problems.

**Comment by vaab** [comment]
I do some nasty things in the ``setup.py`` of those packets, so I'm diving into this to see if this is due to my big tamperings...

**Comment by vaab** [comment]
Some version bissecting indicates that the problem appears between setuptools 9.1 and setuptools 10.0

**Comment by vaab** [comment]
Have left a comment on setuptools issue related to this controversed commit: 

https://bitbucket.org/pypa/setuptools/issue/315/setuptools-should-always-use-its-own

**Comment by vaab** [comment]
Confirming that inserting:

    pip install setuptools==9.1

in ``travis.yml`` will help testing your package. But anyone using setuptools > 9.1 will have trouble installing packets that depends on d2to1 packets.

**Comment by embray** [comment]
I'm going to investigate this.  But what's weird is the only change to setuptools I know about that would relate to this--the one you also posted in--is the one in `setuptools.sandbox.run_setup` that hides away setuptools.  But d2to1 does not use `run_setup` explicitly anywhere, so I wonder why that's happening at all.  pip doesn't use `run_setup`, I don't think.  It just launches a separate process.

**Comment by embray** [comment]
> But anyone using setuptools > 9.1 will have trouble installing packets that depends on d2to1 packets. 

FWIW, I tested with the latest setuptools installing `stsci.convolve` which uses d2to1.  It in turn has some setup_requires dependencies that also use d2to1.  Everything works (although in the output from pip I do have
```
    /internal/1/root/usr/local/lib/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'setup_requires'
      warnings.warn(msg)
```
which is odd, and *possibly* related, though it doesn't seem to actually be a problem...)

**Comment by vaab** [comment]
My generalization was probably a little exagerated. I only tested with a package using d2to1 and having dependency using it also. The failure occurs when installing the dependency not the main package.

I'm currently unable to produce a valid package using only ``setup.py`` (not ``d2to1``) that would correctly require a dependency which is using ``d2to1``. I have a similar warning about ``install_requires`` and it doesn't even try to install my dependency...

**Comment by embray** [comment]
No likely related, but in [this file](https://github.com/0k/kids.test/blob/master/setup.py#L17) the
```python
try:
    from setuptools import setup
except ImportError:
    from distribute_setup import use_setuptools
    use_setuptools()
    from setuptools import setup
```
is not meaningful. `distribute` is deprecated and `distribute_setup` should not be used (the setuptools equivalent is ez_setup.py).  But this won't do anything anyways unless the bootstrap script in question (ez_setup.py) is included in your source distribution.

**Comment by embray** [comment]
Yeah, it seems like a real issue.  The question is how to reproduce it.

**Comment by embray** [comment]
Nevermind about the "unknown distribution option 'setup_requires'" warning I was getting.  It was a [configuration error on my part](https://bitbucket.org/pypa/setuptools/issue/344/typo-in-setuppy-for-setup_requires-entry)

**Comment by embray** [comment]
Okay, I was able to reproduce with installing `kids.sh` from source.

**Comment by embray** [comment]
Closing in preparation for project archival: https://github.com/embray/d2to1/blob/master/README.rst#deprecated  Thanks for all your help with this project!
